JavaScript Final Project

ameerhamzakhalid opened the pull request:
### **Learning objectives**

First place thank you for your future comments, all the comments are welcome and I hope you'll like our work, you can see our video presentation at this [link](https://drive.google.com/file/d/1EqT8P45mNfI40HEgP9Dks8VOMz-HccgV/view?usp=sharing)

Objectives achieved on this Pull Request by @ameerhamzakhalid  and @asifaminisonline 
We use GET request to [Meals DB](https://www.themealdb.com/api.php): API to load the card's information, such as images, titles, IDs, descriptions, etc.
We use POST request to store the data about the likes and comments on each card using the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) and also GET requests to display the saved likes and comments on our project.

- [x] Use JavaScript to make websites dynamic and build basic single-page apps.
- [x] Use ES6 syntax.
- [x] Use ES6 modules.
- [x] Use callbacks and promises.
- [x] Use webpack.
- [x] Apply JavaScript best practices and language style guides in code.
- [x] Use AAA pattern for unit tests.
- [x] Write unit tests for a JavaScript app.
- [x] Follow Gitflow.
- [x] Solve simple git conflicts.
- [x] Send and receive data from an API.
- [x] Use API documentation.
- [x] Understand and use JSON.
- [x] Make JavaScript code synchronous.
- [x] Perform a code review for a team member.

ameerhamzakhalid committed:
added html, js, linters, webpack and jest

We added webpack and jest modules in project
We added html and JavaScript files accordingly
We added linter in project

ameerhamzakhalid committed:
Merge pull request #21 from ameerhamzakhalid/setup-webpack-and-jest

added html, js, linters, webpack and jest

ameerhamzakhalid committed:
added header

We added header in project

ameerhamzakhalid committed:
Merge pull request #22 from ameerhamzakhalid/adding-header

added header

ameerhamzakhalid committed:
added footer

We added footer

ameerhamzakhalid committed:
Merge pull request #23 from ameerhamzakhalid/adding-footer

added footer

ameerhamzakhalid committed:
added display list

I added display list foods feature according which I updated files discussed below:
1)html
2)css
3)javascript

ameerhamzakhalid committed:
Merge pull request #26 from ameerhamzakhalid/display-list-foods

added display list

asifaminisonline committed:
added display comments popup

asifaminisonline committed:
Merge pull request #27 from ameerhamzakhalid/comment-popup-structure

added display comments popup

ameerhamzakhalid committed:
Add counter of items's the API selected

I added counter Item of API
I added navigation to deploy different types of foods

ameerhamzakhalid committed:
Merge pull request #28 from ameerhamzakhalid/add-items-counter

Add counter of items's the API selected

asifaminisonline committed:
added display comments

asifaminisonline committed:
Merge pull request #29 from ameerhamzakhalid/display-comments

Added display comments

ameerhamzakhalid committed:
Add new like

I added new like feature to record user interactions likes

ameerhamzakhalid committed:
Merge pull request #32 from ameerhamzakhalid/add-new-like

Add new like

ameerhamzakhalid committed:
displayed number of likes

ameerhamzakhalid committed:
Merge pull request #33 from ameerhamzakhalid/display-number-likes

displayed number of likes

asifaminisonline committed:
added new comment functionality

asifaminisonline committed:
Added comments counter functionallity

asifaminisonline committed:
Added tests for comments counter

asifaminisonline committed:
Merge pull request #36 from ameerhamzakhalid/comments-counter-tests

Comments counter tests

asifaminisonline committed:
Updated README.md

asifaminisonline committed:
Updated README.md

asifaminisonline committed:
Updated README.md

asifaminisonline committed:
Merge pull request #37 from ameerhamzakhalid/comments-counter-tests

Comments counter tests

ameerhamzakhalid committed:
Add final details

ameerhamzakhalid committed:
Merge pull request #38 from ameerhamzakhalid/final-touch

Add final details

ameerhamzakhalid committed:
update readme

ameerhamzakhalid committed:
update readme

ameerhamzakhalid committed:
update readme

I updated readme file

youmari commented on src/modules/likeDataApi.js:
```diff
@@ -0,0 +1,24 @@
+const projectID = 'cq8n4ydnTGepOHztUKmY';
+const likesLink = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${projectID}/likes/`;
+
+export const addLikeToItem = async (itemId) => {
+  const request = new Request(likesLink);
+  const response = await fetch(request, {
+    method: 'POST', // *GET, POST, PUT, DELETE, etc.
+    mode: 'cors', // no-cors, *cors, same-origin
+    body: JSON.stringify({ item_id: `${itemId}` }),
+    cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+    credentials: 'same-origin', // include, *same-origin, omit
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return response;
+};
+
+export const getLikesToItems = async () => {
+  const response = await fetch(likesLink);
+  const data = await response.json();
+  return data;
+};
```
- [ ] the is looking as expected but this is a small issue in the like feature which is when I click on like the first time it doesn't respond until I click the second click.
this is a bug that may confuse the user so please make sure to fix it

youmari reviewed (changes requested):
Hi @ameerhamzakhalid @asifaminisonline , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ameerhamzakhalid committed:
update getDataFood.js

I updated the getDataFood.js to meet the required changes suggested by the code reviewer

ameerhamzakhalid committed:
fixed linter

I fixed linter error

ameerhamzakhalid committed:
Fixed linter error

I fixed es lint linter error

AmaduKamara commented on src/modules/getDataFood.js:
```diff
@@ -0,0 +1,74 @@
+    item.innerHTML = `
+        <div class="card-title">
+          <h5>${food.strMeal}</h5>
+        </div>        
+        <button type="button" class="btn-recipe">Recipe</button>
+        <div>        
```
[ Optional Change ]

- [ ] Kindly edit the button to have the text "Comment" instead of Recipe to match the wireframe and requirements of this project. Basically, that button is meant for making comment on food and hence, the text on the button should be descriptive of what feature lies in there, which is making comment on a food clicked. 👍🏽

AmaduKamara reviewed (approved):
# Status: Approved 🟢 🟢 🟢 

Hi Team,

Good job on fixing the change highlighted by the previous reviewer 👍🏽 

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ App is nicely styled 👍🏽
✅ Good descriptive PR 👍🏽
✅ All features work fine as expected 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
